shortc.js

```diff
@@ -2,43 +2,60 @@
 const yargs = require('yargs');
 const pkg = require('./package.json');
 const commands = require('./commands.js');
+const chalk = require('chalk');
 
-yargs.version(pkg.version);
+const EXIT_ERROR = 1;
 
-yargs.command({
-  command: 'add',
-  describe: 'Add command',
-  builder: {
-    com: {
-      type: 'string',
-      demandOption: true,
-      describe: 'Add shortcut name',
+const run = async state => {
+  yargs.version(pkg.version);
+
+  yargs.command({
+    command: 'add',
+    describe: 'Add command',
+    builder: {
+      com: {
+        type: 'string',
+        demandOption: true,
+        describe: 'Add shortcut name',
+      },
+      desc: {
+        type: 'string',
+        demandOption: true,
+        describe: 'Add shortcut description',
+      },
     },
-    desc: {
-      type: 'string',
-      demandOption: true,
-      describe: 'Add shortcut description',
+    handler({ com, desc }) {
+      state.addCommand({ com, desc });
     },
-  },
-  handler({ com, desc }) {
-    commands.addCommand(com, desc, commands.pathFile);
-  },
-});
+  });
 
-yargs.command({
-  command: 'read',
-  describe: 'Check all commands',
-  handler() {
-    commands.readCommand(commands.pathFile);
-  },
-});
+  yargs.command({
+    command: 'read',
+    describe: 'Check all commands',
+    handler() {
+      console.dir(state.commands);
+    },
+  });
 
-yargs.command({
-  command: 'path',
-  describe: 'Show path for file with saved commands',
-  handler() {
-    commands.getPath();
-  },
-});
+  yargs.command({
+    command: 'path',
+    describe: 'Show path for file with saved commands',
+    handler() {
+      const path = commands.getSavePath();
+      const colorfullPath = chalk.bgCyan.bold(path);
+      console.log(chalk.blue(`Db is located at ${colorfullPath}`));
+    },
+  });
 
-yargs.parse();
+  yargs.parse();
+};
+
+(async () => {
+  const path = commands.getSavePath();
+  const state = await commands.ShortcState.loadFrom(path);
+  await run(state);
+  return state.saveTo(path);
+})().catch(err => {
+  console.error(`Error: ${err.message}`);
+  process.exitCode = EXIT_ERROR;
```

Comment:
Nice! Instead of immediately terminating, we first finish all tasks, and set exit code to nonzero. That's good!